src/Storage/StorageClient.php

```diff
@@ -217,4 +217,32 @@ public function createBucket($name, array $options = [])
         $response = $this->connection->insertBucket($options + ['name' => $name, 'project' => $this->projectId]);
         return new Bucket($this->connection, $name, $response);
     }
+
+    /**
+     * Registers this StorageClient as the handler for stream reading/writing.
+     *
+     * @param string $protocol The name of the protocol to use. Defaults to
+     *        'gs'.
+     * @throws \RuntimeException
+     */
+    public function registerAsStreamWrapper(string $protocol = null)
+    {
+        if (StreamWrapper::register($protocol)) {
```

Comment:
I like this idea. I can also make the storage an array keyed by the protocol in case you want to register gs:// for one storage client and gs2:// for another.